common/ast/src/ast/statement.rs

```diff
@@ -1134,6 +1137,64 @@ impl<'a> Display for Statement<'a> {
                     }
                 }
             }
+            Statement::Revoke(GrantStatement { source, principal }) => {
+                write!(f, "REVOKE")?;
+                match source {
+                    GrantSource::Role { role } => write!(f, " ROLE {role}")?,
+                    GrantSource::Privs { privileges, level } => {
+                        write!(
+                            f,
+                            " {}",
+                            privileges
+                                .iter()
+                                .map(|p| p.to_string())
+                                .collect::<Vec<_>>()
+                                .join(", ")
+                        )?;
+                        write!(f, " ON")?;
+                        match level {
+                            GrantLevel::Global => write!(f, " *.*")?,
+                            GrantLevel::Database(database_name) => {
+                                if let Some(database_name) = database_name {
+                                    write!(f, " {database_name}.*")?;
+                                } else {
+                                    write!(f, " *")?;
+                                }
+                            }
+                            GrantLevel::Table(database_name, table_name) => {
+                                if let Some(database_name) = database_name {
+                                    write!(f, " {database_name}.{table_name}")?;
+                                }
+                            }
+                        }
+                    }
+                    GrantSource::ALL { level, .. } => {
+                        write!(f, " ALL PRIVILEGES")?;
+                        write!(f, " ON")?;
+                        match level {
+                            GrantLevel::Global => write!(f, " *.*")?,
+                            GrantLevel::Database(database_name) => {
+                                if let Some(database_name) = database_name {
+                                    write!(f, " {database_name}.*")?;
+                                } else {
+                                    write!(f, " *")?;
+                                }
+                            }
+                            GrantLevel::Table(database_name, table_name) => {
+                                if let Some(database_name) = database_name {
+                                    write!(f, " {database_name}.{table_name}")?;
+                                }
```

Comment:
Yes. the sql will error like this:

```
grant all on *.c to xxx;
revole all on *.c to xxx;
```

because not support this usage.

Reply:
if so, Identifier is a better type for database_name instead of Optipn<Identifier>

Reply:
> if so, `Identifier` is a better type for database_name instead of `Option<Identifier>`

BTW, I think it should be Option<String>. Like this sql:

```sql
grant all on t1 to 'user'; -- that means grant all privileges about `current_catalog.current_database.t1` to 'user'
```